File: practice2/intentfilter/src/main/java/ru/mirea/harin/intentfilter/MainActivity.java
package ru.mirea.harin.intentfilter;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;



public class MainActivity extends AppCompatActivity {

    private Button bt1, bt2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bt1 = (Button) findViewById(R.id.button1);

        View.OnClickListener buttont1 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Uri address = Uri.parse("https://www.mirea.ru/");
                Intent openLinkIntent = new Intent(Intent.ACTION_VIEW, address);
                if (openLinkIntent.resolveActivity(getPackageManager()) != null) {
                    startActivity(openLinkIntent);
                } else
                    Log.d("Intent", "Не получается обработать намерение!");
            }
        };
        bt1.setOnClickListener(buttont1);


        bt2 = (Button) findViewById(R.id.button2);

        View.OnClickListener buttont2 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent shareIntent = new Intent(Intent.ACTION_SEND);
                shareIntent.setType("text/plain");
                shareIntent.putExtra(Intent.EXTRA_SUBJECT, "MIREA");
                shareIntent.putExtra(Intent.EXTRA_TEXT, "ФАМИЛИЯ ИМЯ ОТЧЕСТВО");
                startActivity(Intent.createChooser(shareIntent, "МОИ ФИО"));
            }
        };
        bt2.setOnClickListener(buttont2);

    }

}

File: practice1/app/src/main/java/ru/mirea/harin/practice1/MainActivity.kt
package ru.mirea.harin.practice1

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    private var tvOut: TextView? = null
    private var OK: Button? = null
    private var Cancel: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvOut = findViewById<View>(R.id.tvOut) as TextView
        OK = findViewById<View>(R.id.btnOk) as Button
        Cancel = findViewById<View>(R.id.btnCancel) as Button
        val BtnOK =
            View.OnClickListener { tvOut!!.text = "You've clicked OK button" }
        OK!!.setOnClickListener(BtnOK)
        val BtnCancel =
            View.OnClickListener { tvOut!!.text = "You'vw clicked CANCEL button" }
        Cancel!!.setOnClickListener(BtnCancel)

    }

}